### PR TITLE
Clarify user policy

### LINKED ISSUE #34: Update table: users
- no changes to schema

##### Permissions by role

- public: C (register)
- registered, no role: RU (own record only)
- analyst: RU (own record only)
- manager: R (all user records), U (own record only)
- admin: RU ~D~ (all user records)

### PR DESCRIPTION
**Pull request by parndt**
Fixes #34 

- no changes to schema

##### Permissions by role

- public: C (register)
- registered, no role: RU (own record only)
- analyst: RU (own record only)
- manager: R (all user records), U (own record only)
- admin: RU ~D~ (all user records)

### CODE REVIEW

File: app/policies/user_policy.rb [hunk 1]
```diff
@@ -3,23 +3,24 @@ def index?
...
-    return true if @record.id == @user.id
-    return true if @user.role?("admin")
-    @user.role?("manager") && !(@record.role?("admin") || @record.role?("manager"))
+    @user.role?("admin") || @record.id == @user.id
   end
 
   def destroy?
-    return true if @user.role?("admin")
-    @record == @user
+    @user.role?("admin")
```
Comment (tmfrnz) [inline app/policies/user_policy.rb:23]: we should best not allow this, as users will still be referenced throughout the app (i.e. `created_by_id`, `updated_by_id`) to deleted users. 

instead admins can just demote users (strip their role associations) when they no longer need access

File: spec/controllers/users_controller_spec.rb [hunk 2]
```diff
@@ -187,10 +227,10 @@
         expect(subject).to be_not_found
       end
 
-      it "will allow a user to delete themselves" do
+      it "will not allow a user to delete themselves" do
```
Comment (tmfrnz) [inline spec/controllers/users_controller_spec.rb:230]: no one can delete anyone

File: app/policies/user_policy.rb (hunk 1 above)
Comment (parndt) [reply app/policies/user_policy.rb:23]: @tmfrnz OK makes sense - I'll update the specs to ensure nobody can delete anybody and the demotion can happen through the roles functionality instead.

**Comment by parndt** [comment]
@tmfrnz how is it looking now?

**Review by tmfrnz — APPROVED** [review]

**Review by tmfrnz — APPROVED** [review]

**Pull request merged**